hy/lex/parser.py

```diff
@@ -22,84 +22,85 @@
 
 def set_boundaries(fun):
     @wraps(fun)
-    def wrapped(p):
+    def wrapped(state, p):
         start = p[0].source_pos
         end = p[-1].source_pos
-        ret = fun(p)
-        ret.start_line = start.lineno
+        ret = fun(state, p)
+        ret.start_line = start.lineno + state.lineno_offset
         ret.start_column = start.colno
         if start is not end:
             ret.end_line = end.lineno
             ret.end_column = end.colno
         else:
-            ret.end_line = start.lineno
+            ret.end_line = start.lineno + state.lineno_offset
             ret.end_column = start.colno + len(p[0].value)
+        ret.end_line += state.lineno_offset
         return ret
     return wrapped
 
 
 def set_quote_boundaries(fun):
     @wraps(fun)
-    def wrapped(p):
+    def wrapped(state, p):
         start = p[0].source_pos
-        ret = fun(p)
-        ret.start_line = start.lineno
+        ret = fun(state, p)
+        ret.start_line = start.lineno + state.lineno_offset
         ret.start_column = start.colno
-        ret.end_line = p[-1].end_line
+        ret.end_line = p[-1].end_line + state.lineno_offset
         ret.end_column = p[-1].end_column
         return ret
     return wrapped
 
 
 @pg.production("main : list_contents")
-def main(p):
+def main(state, p):
     return p[0]
 
 
 @pg.production("main : $end")
-def main_empty(p):
+def main_empty(state, p):
     return []
 
 
-def reject_spurious_dots(*items):
-    "Reject the spurious dots from items"
-    for list in items:
-        for tok in list:
-            if tok == "." and type(tok) == HySymbol:
-                raise LexException("Malformed dotted list",
-                                   tok.start_line, tok.start_column)
+# def reject_spurious_dots(*items):
+#     "Reject the spurious dots from items"
+#     for list in items:
+#         for tok in list:
+#             if tok == "." and type(tok) == HySymbol:
+#                 raise LexException("Malformed dotted list",
+#                                    tok.start_line, tok.start_column)
```

Comment:
Lots of commented-out code here.

Reply:
Yeah, I wasn't sure where/if that was ever used, so I didn't update it and only commented it out.

Reply:
Oh yeah, looks like it's dead code. I vaguely remember making it obsolete when editing the lexer a while ago. Why not just add another commit to this PR, rebased to before the other commits, deleting it?